TESTS/mbed_platform/minimal-printf/compliance/main.cpp

```diff
@@ -42,6 +42,94 @@
 
 using namespace utest::v1;
 
+#if defined(__NEWLIB_NANO)
+
+#define MAX_STRING_SIZE 100
+//Global buffer used by test cases to store string literal
+char expected_string[MAX_STRING_SIZE];
+
+typedef enum {
+    RADIX_10,//radix base for decimal
+    RADIX_16 //radix base for Hexadecimal
+} radix_base;
+
+// This function is similar to itoa but it can convert any data type with decimal and hexadecimal numerical value into a string
+static void convert_to_string(unsigned long long  value,  char str[],  radix_base radix = RADIX_10, bool sign = false)
+{
+    char        buffer[MAX_STRING_SIZE];
+    char       *dest = buffer + sizeof(buffer);
+
+    // If value is 0
+    if (value == 0) {
+        memcpy(str, "0", 2);
+        return;
+    }
+
+    if (sign) {
+        if ((long long) value < 0) {
+            // Make value postive
+            value = -value;
+        }
+    }
+    *--dest = '\0';
+    switch (radix) {
+        case RADIX_16:
+            while (value) {
+                * --dest = '0' + (value & 0xF);
+                if (*dest > '9') {
+                    *dest += 'A' - '9' - 1;
+                }
+                value >>= 4;
+            }
+            break;
+        case RADIX_10:
+            while (value) {
+                *--dest = '0' + (value % 10);
+                value /= 10;
+            }
+            break;
+
+        default:
+            MBED_ASSERT(0);
+            break;
+    }
+
+    if (sign) {
+        // Add the negative sign for negative numerical value
+        *--dest = '-';
+    }
+    // Copy the converted string from dest into str buffer of converted string length
+    memcpy(str, dest, buffer + sizeof(buffer) - dest);
+}
+
+//This function used to calculate the string length on format 'prefix + data + suffix'
+static int expected_string_length(char *prefix = nullptr, long long value, radix_base base, char *suffix = nullptr, bool sign = false)
+{
+    char *exp_str = &expected_string[0];
+    int prefix_length = 0;
+    memset(exp_str, 0, MAX_STRING_SIZE);
+    if (prefix) {
+        prefix_length = strlen(prefix);
+        strncpy(exp_str, prefix, prefix_length);
+    }
+    convert_to_string(value, &exp_str[prefix_length], base, sign);
+    if (suffix) {
+        strcat(exp_str, suffix);
+    }
+    return strlen(exp_str);
+}
+
+//This function used to extract the prefix format from the fmt
+static void extract_prefix(const char *fmt, char *prefix = nullptr)
```

Comment:
I don't think it makes sense to have a default argument of nullptr. This function should not be called with nullptr. In fact you should assert if `prefix` is null.

Reply:
My comment above is still valid.

Why not replace this function with:
```
int i = 0;
while (fmt && prefix && fmt[i] != '%') {
        prefix[i] = fmt[i];
        i++;
}
```